src/validation.cpp

```diff
@@ -5117,6 +5168,149 @@ bool ChainstateManager::PopulateAndValidateSnapshot(
     return true;
 }
 
+// Currently, this function holds cs_main for its duration, which could be for
+// multiple minutes due to the GetUTXOStats call. This hold is necessary
+// because we need to avoid advancing the background validation chainstate
+// farther than the snapshot base block - and this function is also invoked
+// from within ConnectTip, i.e. from within ActivateBestChain, so cs_main is
+// held anyway.
+//
+// Eventually (TODO), we could somehow separate this function's runtime from
+// maintenance of the active chain, but that will either require
+//
+//  (i) setting `m_disabled` immediately and ensuring all chainstate accesses go
+//      through IsUsable() checks, or
+//
+//  (ii) giving each chainstate its own lock instead of using cs_main for everything.
+SnapshotCompletionResult ChainstateManager::MaybeCompleteSnapshotValidation(
```

Comment:
It looks like there has been a rebase error and two previous commits were combined into one commit that doesn't have a valid commit description. The two previous commits were:

- "move-only-ish: init: factor out chainstate initialization" (aead772a24b1f7759cd756bf6534a70dc41ca080)
- "validation: add ChainMan logic for completing UTXO snapshot validation" (863e5f389e08cd2069e4834df82879ec39142aac)

and the new combined commit is 

 - "move-only-ish: init: factor out chainstate initialization" (07b9dea9d4e8de2ce22e8ba50172a4af85801931)

It would help review to split these up and separate moveonly changes from other changes

Reply:
Wow, no idea how that got so screwed up. Will fix, thanks.